### PR TITLE
[#IOPID-1201] Disable notifications when the user locks the session from ioapp.it

### PR DESCRIPTION
**Pull request by gquadrati**
#### List of Changes

- add deleteInstallation to lockUserAuthentiction flow](https://github.com/pagopa/io-backend/commit/fa4308d5be855b9917a2d8e98dbf0ea2f680c01b)
- add tests](https://github.com/pagopa/io-backend/commit/d6eacbdc6a825b6e7ee8926f977906650596a434)

#### Motivation and Context

Avoid to sent notifications to the user's device when a session lock occurred

#### How Has This Been Tested?

Unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/sessionLockController.ts [hunk 1]
```diff
@@ -414,6 +417,35 @@ export default class SessionLockController {
...
+        () =>
+          this.notificationServiceFactory(fiscalCode).deleteInstallation(
+            fiscalCode
+          ),
+        (err) =>
+          Error(
+            `Cannot delete Notification Installation: ${JSON.stringify(err)}`
+          )
+      ),
+      TE.chainW(
```
Comment (arcogabbo) [inline src/controllers/sessionLockController.ts:433]: widen not needed. here we are returning `Error` in the left side as the step before
```suggestion
      TE.chain(
```

**Review by arcogabbo — APPROVED** [review]

File: src/controllers/sessionLockController.ts [hunk 2]
```diff
@@ -245,6 +247,7 @@ export default class SessionLockController {
             pipe(
               AP.sequenceT(TE.ApplicativeSeq)(
                 ...this.buildInvalidateUserSessionTask(fiscalCode),
+                this.clearInstallation(fiscalCode),
```
Comment (BurnedMarshal) [inline src/controllers/sessionLockController.ts:250]: If this operation fail, we cannot lock the user session.

**Review by BurnedMarshal — APPROVED** [review]
LGTM

File: src/controllers/sessionLockController.ts (hunk 2 above)
Comment (gquadrati) [reply src/controllers/sessionLockController.ts:252]: Added a comment to clarify the reasoning behind the chosen order. Thanks. [569ee4c](https://github.com/pagopa/io-backend/pull/1080/commits/569ee4c8084032a3cfe6388fdf85f719adb6f788)

**Pull request merged**